source/llassetgen/include/llassetgen/Atlas.h

```diff
@@ -0,0 +1,54 @@
+#pragma once
+
+#include <llassetgen/DistanceTransform.h>
+#include <llassetgen/Image.h>
+#include <llassetgen/Packing.h>
+
+namespace llassetgen {
+    namespace internal {
+        template <class Iter>
+        constexpr int checkImageIteratorType() {
+            using IterTraits = typename std::iterator_traits<Iter>;
+            using IterRefType = typename IterTraits::reference;
+            using IterCategory = typename IterTraits::iterator_category;
+            static_assert(std::is_assignable<Image, IterRefType>::value, "Input elements must be assignable to Image");
```

Comment:
Does this still work with the copy constructor removed? My guess is that this would check if can assign an `Image&` to an `Image`, which would only work with a copy. Otherwise we can always use `std::is_assignable<Image&, IterRefType>`.

Reply:
Good point. Unfortunately `std::is_assignable<Image&, IterRefType>` fails as well, though.

Reply:
Maybe replace `Image&` by `const Image&`? Or just use `IterTraits::value_type` instead of `::reference` (can't test this right now unfortunately).